martin-core/src/resources/sprites/mod.rs

```diff
@@ -71,27 +71,165 @@ impl SpriteSources {
 
     /// Adds a sprite source directory containing SVG files.
     /// Files are ignored - only directories accepted. Duplicates ignored with warning.
-    pub fn add_source(&mut self, id: String, path: PathBuf) {
+    /// Performs basic validation of SVG format before adding the source.
+    pub fn add_source(&mut self, id: String, path: PathBuf) -> Result<(), SpriteError> {
         let disp_path = path.display();
+
         if path.is_file() {
-            warn!("Ignoring non-directory sprite source {id} from {disp_path}");
-        } else {
-            match self.0.entry(id) {
-                Entry::Occupied(v) => {
-                    warn!(
-                        "Ignoring duplicate sprite source {} from {disp_path} because it was already configured for {}",
-                        v.key(),
-                        v.get().path.display()
-                    );
+            return Err(SpriteError::DirectoryValidationFailed(
+                path,
+                "Path is not a directory".to_string(),
```

Comment:
a minor nit, but makes for a better pattern -- if you have several similar errors with some magical string, easier to just have dedicated errors one for each. This way they are much faster to create (you don't need to allocate a new string on each error), and all content of an error is in the error description only